samcli/commands/build/command.py

```diff
@@ -136,6 +136,15 @@
     help="Enabled parallel builds. Use this flag to build your AWS SAM template's functions and layers in parallel. "
     "By default the functions and layers are built in sequence",
 )
+@click.option(
+    "--mount-with-write",
```

Comment:
For boolean options, we usually provide both values. So we can rename this to; `--mount-with-write/--mount-with-read`? Default value would be `--mount-with-read` which will resolve as `mount_with_write=False` down below.